src/tools/robot-utils.cpp

```diff
@@ -5,28 +5,92 @@
  *
  */
 
-#include <boost/make_shared.hpp>
-#include <boost/shared_ptr.hpp>
+#ifdef BOOST_MPL_LIMIT_VECTOR_SIZE
+#pragma push_macro("BOOST_MPL_LIMIT_VECTOR_SIZE")
+#undef BOOST_MPL_LIMIT_VECTOR_SIZE
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/xml_parser.hpp>
+#pragma pop_macro("BOOST_MPL_LIMIT_VECTOR_SIZE")
+#else
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/xml_parser.hpp>
+#endif
+
 #include <dynamic-graph/factory.h>
 #include <iostream>
+#include <sstream>
 #include <sot/core/debug.hh>
 #include <sot/core/robot-utils.hh>
 
 namespace dynamicgraph {
 namespace sot {
 namespace dg = ::dynamicgraph;
+namespace pt = boost::property_tree;
 using namespace dg;
 using namespace dg::command;
 
-RobotUtil VoidRobotUtil;
 ForceLimits VoidForceLimits;
 JointLimits VoidJointLimits;
 Index VoidIndex(-1);
 
 RobotUtilShrPtr RefVoidRobotUtil() {
-  return boost::make_shared<RobotUtil>(VoidRobotUtil);
+  return std::shared_ptr<RobotUtil>(nullptr);
+}
+
+ExtractJointMimics::ExtractJointMimics(std::string & robot_model) {
+  // Parsing the model from a string.
+  std::istringstream iss(robot_model);
+  boost::property_tree::read_xml(iss,tree_);
+  go_through_full();
+}
+
+const std::vector<std::string> &
+ExtractJointMimics::get_mimic_joints() {
+  return mimic_joints_;
 }
 
+void ExtractJointMimics::go_through_full()  {
+  current_joint_name_="";
+  go_through(tree_,0,0);
+}
+
+void ExtractJointMimics::go_through(pt::ptree &pt, int level, int stage)
+{
```

Comment:
Might be worth adding a comment on what happens in this recursive function.

Reply:
Done in commit f7f8ebb